### PR TITLE
Caron, Lessard, Wu, Poisot

### PR DESCRIPTION
**Pull request by tpoisot**
**AUTHOR**

Dear @ReScience/editors,

I request a review for the following replication:

### Original article

**Title:**  Insect natural enemies as regulating factors
**Author(s):**  M.P. Hassell
**Journal (or Conference):** Journal of Animal Ecology  
**Year:**  1985
**DOI:**  https://doi.org/10.2307/4641
**PDF:**   

### Replication

**Author(s)**:  @DominiqueCaron @vincelessard @QileWu @tpoisot
**Repository**:  https://github.com/BIO6032/2018_replication_hassell_1985 
**PDF**:  https://github.com/BIO6032/2018_replication_hassell_1985/blob/master/article/Caron-Lessard-Wu-Poisot-2018.pdf
**Keywords**:  population dynamics, ecology, stochastic models
**Language**:  Julia
**Domain**:  ecology

### Results

* [x] Article has been fully replicated
* [ ] Article has been partially replicated
* [ ] Article has not been replicated

### Potential reviewers

---

**EDITOR**

* [x] Editor acknowledgment (@pdebuyl `9 June 2018`)
* [x] Reviewer 1 (@ChrisRackauckas `13 June 2018`)
* [x] Reviewer 2 (@jsta `14 June 2018`)
* [x] Review 1 decision: accept `12 October 2018`
* [x] Review 2 decision: accept `20 February 2019`
* [x] Editor decision [accept/reject] `13 March 2019`

### CODE REVIEW

**Comment by rougier** [comment]
Thanks for the submission @tpoisot. 
@karthik Can you handle this new submission ? Else, can you point to a potential editor (from the board)?

We'll need more ecology editors at some point...

**Comment by tpoisot** [comment]
@rougier should we open an issue for additional editors? I have a few ideas for ecologists.

**Comment by rougier** [comment]
Maybe it's better to contact me & @khinsen via email.

**Comment by rougier** [comment]
@karthik Gentle reminder

**Comment by karthik** [comment]
@rougier Unfortunately my editorial load (across various journals) Is high at the moment and I wont be able to handle this. Please follow up with another editor.

**Comment by rougier** [comment]
@karthik Ok, thanks for the answer.  @tpoisot Do you have some ideas for the editor, can you contact me by mail?

**Comment by rougier** [comment]
Still looking for an editor...

**Comment by pdebuyl** [comment]
@rougier I can do it

**Comment by rougier** [comment]
Thanks @pdebuyl. Do you feel confortable enough with the topic for finding reviewers?

**Comment by pdebuyl** [comment]
I have one Julia expert in mind and have ideas for a second one in ecology from the list of reviewers of ReScience. As far as dynamical systems go, I feel at home :-) I'll get to this on monday morning.

**Comment by pdebuyl** [comment]
Just for info, I have started to contact reviewers. I will update the PR when I have a confirmation.

**Comment by pdebuyl** [comment]
@tpoisot can you add the DOI of the original paper? Do you have a accessible pdf of it?

**Comment by tpoisot** [comment]
@pdebuyl there is no DOI for the paper, and the PDF is not open access so I cannot post a copy here

**Comment by rougier** [comment]
Actually there is a DOI since you give it: 10.2307/4641
(just notice Pierre added it...)

**Comment by tpoisot** [comment]
Nice! I had no idea this had a DOI.

**Comment by pdebuyl** [comment]
I wanted to find a pdf with google scholar and found the doi. There is a pdf on jstor (you are supposed to log in but with noscript I could get the pdf with a click).

**Comment by pdebuyl** [comment]
@jsta @ChrisRackauckas first ping

File: code/functions.jl [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+- `c`: Number of parasitoids emerging from each host parasitized
+- `a`: Searching efficiency (per capita)
+- `h`: Saturation number of parasitoids
+- `b`: Rate of appoaching h
+- `th`: Handling time (as a proportion of the total time)
+- `m`: Extent of clumping of the parasitoid attacks
+
+Return : `dynamics`: Matrix containing the host and parasite density for the initial populations and for every generation simulated
+"""
+function simulation(N::Float64, P::Float64; t::Int64=50, f=specialist_dyn, F=4.0, D=0.5, c=1.0, a=0.5, h=10.0, b=25.0, th=0.0, m=0.2, D_sd=0.0, a_sd=0.0, h_sd=0.0, c_sd=0.0)
```
Comment (ChrisRackauckas) [inline code/functions.jl]: I would split to keyword arguments at `F`. A long list of unnamed parameters can be hard to parse.

File: code/functions.jl [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+"""
+***Generalist parasitoids population size(Initial population size of female parasitoids) ***
+
+- `h`: Saturation number of parasitoids
+- `b`: Rate of appoaching h
+- `N`: Initial host population size
+
+Return : `P`: Generalist parasitoids population size
+"""
+function generalist_dyn(N::Float64, P::Float64, p)
```
Comment (ChrisRackauckas) [inline code/functions.jl]: `P` is unused.

File: code/functions.jl [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+
+"""
+***Generalist parasitoids population size(Initial population size of female parasitoids) ***
+
+- `h`: Saturation number of parasitoids
+- `b`: Rate of appoaching h
+- `N`: Initial host population size
+
+Return : `P`: Generalist parasitoids population size
```
Comment (ChrisRackauckas) [inline code/functions.jl]: Return is `Pt`?

File: code/functions.jl [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+- `F`: Finite rate of increase of the host population
+- `D`: Density independent mortality (as a probability of survival)
+- `c`: Number of parasitoids emerging from each host parasitized
+- `a`: Searching efficiency (per capita)
+- `h`: Saturation number of parasitoids
+- `b`: Rate of appoaching h
+- `th`: Handling time (as a proportion of the total time)
+- `m`: Extent of clumping of the parasitoid attacks
+
+Return : `dynamics`: Matrix containing the host and parasite density for the initial populations and for every generation simulated
```
Comment (ChrisRackauckas) [inline code/functions.jl]: should say the row/column structure of the return (i.e. time is rows).

File: code/functions.jl [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    # Iterations
+    for current_time in 1:(t+1)
+        current_D=generate(D, D_sd)
+        current_a=generate(a, a_sd)
+        current_h=generate(h, h_sd)
+        current_c=generate(c, c_sd)
+        current_p = @NT(F=F, D=current_D, c=current_c, a=current_a, h=current_h, b=b, th=th, m=m)
+        N_next, P_next = timestep(dynamics[current_time,2], dynamics[current_time,3], current_p; parasite_dyn=f)
+        # Population sizes
+        dynamics[current_time+1,1] = current_time
```
Comment (ChrisRackauckas) [inline code/functions.jl:45]: `dynamics` is structured an used in a row-major way here.

File: code/functions.jl [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+        dynamics[current_time+1,3] = P_next
+        # Real parameter values
+        dynamics[current_time,4] = current_D
+        dynamics[current_time,5] = current_a
+        dynamics[current_time,6] = current_h
+        dynamics[current_time,7] = current_c
+    end
+    p = @NT(F=F, D=D, c=c, a=a, h=h, b=b, th=th, m=m)
+    # Return
+    return dynamics[1:(end-1),:], p
```
Comment (ChrisRackauckas) [inline code/functions.jl:56]: Another row-major operation. It makes me wonder if this should've been transposed.

File: code/test.jl [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+cor_a = zeros(Float64, N)
+cor_D = zeros(Float64, N)
+cor_c = zeros(Float64, N)
+
+for i in eachindex(cor_a)
+    sd = i == 1 ? 0.0 : 0.5
+    sim, params = simulation(50.0, 25.0, m=0.2, F=4.0, D=0.5, c=1.0, a=0.5, th= 0.0, m=0.5, f=specialist_dyn, a_sd=sd)
+    kval = mapslices((r) -> kvalue_by_generation(r, params), sim, 2)
+    a, b = linreg(vec(log10.(sim[:,2])), vec(kval))
```
Comment (ChrisRackauckas) [inline code/test.jl]: `linreg` is being removed from Base so even though this is a Julia v0.6.2 code, I think it should strive to have v1.0 readability. I would suggest swapping out `linreg` for the same thing as a `\`.

File: code/test.jl [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+pl_a = density(cor_a[2:end], xlim=(0,0.6), ylim=(0,10), frame=:origin, c=:black, fill=(0, :grey, 0.2), leg=false)
+vline!(pl_a, [first(cor_a)], lw=2, ls=:dot, c=:grey)
+
+pl_D = density(cor_D[2:end], xlim=(0,0.6), ylim=(0,10), frame=:origin, c=:black, fill=(0, :grey, 0.2), leg=false)
+vline!(pl_D, [first(cor_D)], lw=2, ls=:dot, c=:grey)
+
+pl_c = density(cor_c[2:end], xlim=(0,0.6), ylim=(0,10), frame=:origin, c=:black, fill=(0, :grey, 0.2), leg=false)
+vline!(pl_c, [first(cor_c)], lw=2, ls=:dot, c=:grey)
+
+plot(pl_D, pl_c, pl_a, layout=(3,1))
```
Comment (ChrisRackauckas) [inline code/test.jl:39]: Are there any unit tests that can be applied here instead of all visual?

File: code/run_all.jl [hunk 9]
```diff
@@ -0,0 +1,14 @@
+using NamedTuples
+using Plots
+using StatPlots
+using Measures
+using Distributions
+
+include("functions.jl")
+
```
Comment (ChrisRackauckas) [inline code/run_all.jl:9]: is there a reason why figures 1 and 2 are missing?

File: article/Caron-Lessard-Wu-Poisot-2018.md [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+# Methods
+
+The mathematical formulation used in this paper to show the difficulties of
+detecting natural enemies as regulating factors are the same that were used in
+the original paper by @Hassell85 . First of all, the host population dynamics
+are given as
+
+$$N_{t+1} = F \times N_t \times f(N_t,P_t) \times D$$ {#eq:1}
+
+where $N(*t*)$ and $N(*t+1*)$ represent the host population at generation $t$
```
Comment (ChrisRackauckas) [inline article/Caron-Lessard-Wu-Poisot-2018.md]: _t and _{t+1}?

File: code/figure3.jl [hunk 11]
```diff
@@ -0,0 +1,44 @@
+include("seed.jl")
+
+# Simulation with specialist natural enemies with m=0.2
+sim1, params1 = simulation(50.0, 25.0, m=0.2, F=4.0, D=0.5, c=1.0, a=0.5, th = 0.0)
```
Comment (ChrisRackauckas) [inline code/figure3.jl]: Just pass the values which are different from the default. It looks like all are the same as the default as `m=0.2`, but this isn't clear at a glance.

File: code/figure3.jl [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+   lc=:grey, lw=2,
+   leg=false, frame=:origin,
+   xlims=(-0.5, 2.0),
+   ylims=(0, 1.5))
+xlabel!(fig3c, "Host density (log 10)")
+ylabel!(fig3c, "k-value")
+annotate!(fig3c, -0.25, 1.35, text("(c)"))
+
+# Simulation with specialist natural enemies with m=0.8
+sim2, params2 = simulation(50.0, 25.0, m=0.8, F=4.0, D=0.5, c=1.0, a=0.5, th = 0.0)
```
Comment (ChrisRackauckas) [inline code/figure3.jl]: If only `m` is passed then the comment is clear from the code.

File: code/figure3.jl [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+# Fig 3b)
+fig3b = plot(sim2[:,1],sim2[:,2], label="Hosts", frame=:origin, lw=3, leg=false)
```
Comment (ChrisRackauckas) [inline code/figure3.jl]: here you do use columns, so I guess it's ambiguous which should be preferred (except for performance of course)

File: code/figure4.jl [hunk 14]
```diff
@@ -0,0 +1,25 @@
+include("seed.jl")
+
+# Simulation with generalist natural enemies
+sim, params = simulation(50.0, 25.0, m=0.5, F=4.0, D=0.5, h=10.0, b=25.0, a=0.5,
```
Comment (ChrisRackauckas) [inline code/figure4.jl]: only `f` is changed here?

File: code/figure5.jl [hunk 15]
```diff
@@ -0,0 +1,68 @@
+include("seed.jl")
+
+# Simulation with specialist natural enemies with stochasticity on D
+sim1, params1 = simulation(50.0, 25.0, m=0.2, F=4.0, D=0.5, c=1.0, a=0.5, th= 0.0, m=0.5, f=specialist_dyn, D_sd=0.5)
```
Comment (ChrisRackauckas) [inline code/figure5.jl]: Julia Base uses `std` for standard deviation. It would probably be good to keep the consistency, i.e. `D_std`.

File: code/functions.jl [hunk 16]
```diff
@@ -0,0 +1,166 @@
+"""
+TODO
+"""
+function generate(x, σ)
+    @assert σ >= 0.0
+    return σ > 0.0 ? rand(TruncatedNormal(x,σ, 0.0, 100.0)) : x
```
Comment (ChrisRackauckas) [inline code/functions.jl]: space after `x,`.

File: code/figure6.jl [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+ylabel!(fig6d, "k-value")
+annotate!(fig6d, -0.25, 1.35, text("(d)"))
+
+sim1_R2 = cor(vec(log10.(sim1[:,2])), vec(kval1))^2
+sim1_a, sim1_b = linreg(vec(log10.(sim1[:,2])), vec(kval1))
+
+# Simulation with specialist natural enemies with stochasticity on c
+sim2, params2 = simulation(50.0, 25.0, m=0.5, F=4.0, D=0.5, h=10.0, b=25.0, a=0.5, th= 0.0, m=0.5, f=generalist_dyn, h_sd=5)
+# Fig 5b)
+fig6b = plot(sim2[:,1],sim2[:,2], label="Hosts", frame=:origin, lw=3, leg=false, ylims=(0, 90))
```
Comment (ChrisRackauckas) [inline code/figure6.jl]: every plot command is missing the space after the `,` of the first arg.

**Comment by ChrisRackauckas** [comment]
The code is very self-documenting and easy to read. I left a few comments on it and the paper. I'll comb through the details of the replication later this week.

**Comment by pdebuyl** [comment]
Thanks @ChrisRackauckas for the first round of review.

**Comment by jsta** [comment]
## Replication



* The authors claim: 

_For every figure reproduced, we will use the exact same values that were used in the original paper for the different parameters._

However, it seems that for Figure 2 (the original Figure 4) and Figure 4 (the original Figure 6) the authors used a starting host population of 50 instead of 25. I don't think this affects the replication but maybe there is an interesting reason the authors could mention?

## Reproducibility 



* The code took about 4 minutes to run on my machine and ran without error (including module precompilation).

* Rebuilt figures appear identical to the submission figures. 

## Code clarity



* There are some docstrings (`generate()`, and `timestep()`) that read `TODO`. May want to write a minimal description for these.

* I realize that the file name `run_all.jl` is self-explanatory but I still think I would be idea to mention it as the entry point in `code/README.md`. They could also add it to `article/Makefile` but I think this is not strictly necessary.

## Article quality 



* I think the authors did a good job explaining the challenges encountered in reproducing a stochastic result. Adding the supplementary distribution figures was a great idea.

* I also liked how their graphics differed somewhat from the original in that they carried the time-series linking of the host density vs k-values throughout the paper instead of limiting it to Figure 1 (the original Figure 3). 

* Looks like there is some issue with the figures "floating" into the reference list

* There are some cases where the text reads in future tense _we will use_ instead of _we use_ or _we used_.

**Comment by pdebuyl** [comment]
@jsta Thank you for the review!

@ChrisRackauckas ping for the data part :-)

@tpoisot There are already code and paper related queries by the reviewer, you can start to address them or wait for @ChrisRackauckas 's follow-up.

**Comment by pdebuyl** [comment]
@ChrisRackauckas Can you let us know when you will be able to perform the rest of the review?

File: code/test.jl [hunk 18]
```diff
@@ -0,0 +1,52 @@
+using NamedTuples
+using Plots
```
Comment (ChrisRackauckas) [inline code/test.jl]: need Distributions

File: code/figure3.jl [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+xlabel!(fig3b, " ")
+ylabel!(fig3b, " ")
+annotate!(fig3b, 5, 45, text("(b)"))
+
+kval2 = mapslices((r) -> kvalue_by_generation(r, params2), sim2, 2)
+fig3d = plot(log10.(sim2[:,2]), kval2,
+   m=:circle, msc=:white, msw=2, ms=4, mc=:grey,
+   lc=:grey, lw=2,
+   leg=false, frame=:origin,
+   xlims=(-0.5, 2.0),
```
Comment (ChrisRackauckas) [inline code/figure3.jl:37]: these axis limits are far too wide to make it clear that it's a replication. Do the same 1.3 - 1.7 and 0.25 - 0.4

**Comment by ChrisRackauckas** [comment]
## Replication

The authors claim that all of the figures have been replicated. Most discrepancies are mentioned. However, there are still a few things to mention. There is a pretty large discrepancy between Figure 5B and the paper. While it is stochastic, the oscillations are not in the right range and thus this part is not replicated, and it's not clear to me why that is the case. From the files I see that a value c = 1.0 +- 0.5 is used, but in the original paper and the generated figure it mentions c = 0.5 +- 0.5. The original paper does not give a value for c from what I can see in Figure 3 and 4, but since the base value here is what's used in the deterministic simulation figures as the value, it's not clear to me why c=1.0 is used in the Figure 3 and Figure 4 replications instead of c=0.5 (unless I missed something in the original paper). This may explain some of the discrepancies.

Figure 6a and Figure 6b have a similar problem in that the fluctuations are not of the same size as the original figure. This indicates something different in the models which could related to the lack of replication of Figure 6d and 6e. If these results are actually different, this is a failure to replicate and I would like to see some explanation of what's different, i.e. what kind of coding mistakes could reproduce the nice line in Figure 6d.

Figure 7 again uses a different value for c than mentioned in the figure.

## Reproducibility of the Replication

It all generates the same figures as shown in the generated PDF. Code comments have been included, and the tests need to add Distributions.jl to them.

## Code Clarity

The code is very clear and self-documenting. While there are some missing docstrings they don't really matter. However, there are a few issues, like passing unnecessary keyword arguments that match the default values, which hinder the clarity. When passing in optional values, they should only be passed to indicate that they are different from the standard. Since the whole paper uses one base parameter regime, most of the simulations should only be passing one or two parameters, and then it will be very clear what's different in each one.

It doesn't have a useful README though. That should be filled out. Also, is there a license that should be added?

## Article Clarity

The article is a nice and clear read.

**Comment by ChrisRackauckas** [comment]
Oh I second the motion to change the initial populations to match the original paper. That doesn't really change the results but will make the replication more clear.

**Comment by pdebuyl** [comment]
Thanks @ChrisRackauckas !

I did miss the missing license in code and it is indeed necessary for the authors to pick one.

**Comment by pdebuyl** [comment]
@DominiqueCaron @vincelessard @QileWu @tpoisot both reviews are done. The next step is to address their comments.

File: code/functions.jl (hunk 2 above)
Comment (tpoisot) [reply code/functions.jl]: This is admittedly not optimal, but it allows to pass `specialist_dyn` (where it is used) or `generalist_dyn` (where it is not) to the main functions in a single step. We have kept this as is.

File: code/functions.jl (hunk 3 above)
Comment (tpoisot) [reply code/functions.jl]: Correct, this was a typo, fixed in ebf540e4665

File: code/functions.jl (hunk 4 above)
Comment (tpoisot) [reply code/functions.jl]: Fixed.

File: code/functions.jl (hunk 6 above)
Comment (tpoisot) [reply code/functions.jl:56]: Same as above.

File: code/functions.jl (hunk 5 above)
Comment (tpoisot) [reply code/functions.jl:45]: Given the time it takes to run the model, and the fact that replicates are almost never needed, we're OK sacrificing performance if it means keeping the data organised in the way ecologists expect them to. There are a few places throughout where the code is not as optimal as it should, always in the name of clarity (this was done as part of a class).

File: code/test.jl (hunk 7 above)
Comment (tpoisot) [reply code/test.jl]: Good point. Fixed.

File: code/run_all.jl (hunk 9 above)
Comment (tpoisot) [reply code/run_all.jl:9]: The generated figures keep the names they had in the original MS -- Fig 1 and 2 are explanatory diagrams.

Comment (tpoisot) [reply code/run_all.jl:9]: (we added a note to clarify this)

File: article/Caron-Lessard-Wu-Poisot-2018.md (hunk 10 above)
Comment (tpoisot) [reply article/Caron-Lessard-Wu-Poisot-2018.md]: Yes, fixed.

File: code/run_all.jl (hunk 9 above)
Comment (ChrisRackauckas) [reply code/run_all.jl:9]: I did a walkthrough of the code before the paper, that's why I got confused here 👍

File: code/figure4.jl (hunk 14 above)
Comment (tpoisot) [reply code/figure4.jl]: Yes, fixed too.

File: code/figure5.jl (hunk 15 above)
Comment (tpoisot) [reply code/figure5.jl]: Done.

File: code/functions.jl (hunk 16 above)
Comment (tpoisot) [reply code/functions.jl]: Fixed.

File: code/figure6.jl (hunk 17 above)
Comment (tpoisot) [reply code/figure6.jl]: Fixed.

File: code/test.jl (hunk 18 above)
Comment (tpoisot) [reply code/test.jl]: Added.

**Comment by tpoisot** [comment]
@DominiqueCaron I have made most changes to the code, can you start changing the text? In particular, 

> There is a pretty large discrepancy between Figure 5B and the paper. While it is stochastic, the oscillations are not in the right range and thus this part is not replicated, and it's not clear to me why that is the case. From the files I see that a value c = 1.0 +- 0.5 is used, but in the original paper and the generated figure it mentions c = 0.5 +- 0.5. The original paper does not give a value for c from what I can see in Figure 3 and 4, but since the base value here is what's used in the deterministic simulation figures as the value, it's not clear to me why c=1.0 is used in the Figure 3 and Figure 4 replications instead of c=0.5 (unless I missed something in the original paper). This may explain some of the discrepancies.

> Figure 6a and Figure 6b have a similar problem in that the fluctuations are not of the same size as the original figure. This indicates something different in the models which could related to the lack of replication of Figure 6d and 6e. If these results are actually different, this is a failure to replicate and I would like to see some explanation of what's different, i.e. what kind of coding mistakes could reproduce the nice line in Figure 6d.

**Comment by DominiqueCaron** [comment]
@tpoisot Sure! I will start this as soon as possible. Thanks for the changes.

**Comment by pdebuyl** [comment]
Hi @DominiqueCaron is there any update on the manuscript?

**Comment by tpoisot** [comment]
@pdebuyl I need to finish the edits on the test, I will do this soon.

**Comment by pdebuyl** [comment]
@tpoisot ping

**Comment by pdebuyl** [comment]
@DominiqueCaron @vincelessard @QileWu @tpoisot ping

**Comment by tpoisot** [comment]
Sorry about the delay, been a little caught up dealing with :baby: :baby: . I'll update the PR in the second week of october.

**Comment by pdebuyl** [comment]
No worry, I ping to make sure the submission is still alive. Happy parenting :-)

**Comment by tpoisot** [comment]
@pdebuyl here it goes -- the text has been updated by @DominiqueCaron, and the PDF is up to date

**Comment by pdebuyl** [comment]
@jsta can you review the update by the authors?

**Comment by pdebuyl** [comment]
@ChrisRackauckas can you review the update by the authors?

**Comment by pdebuyl** [comment]
@tpoisot @DominiqueCaron thank you for the update. 

May you add doi's in the bibfile whenever possible? This should then render with links in the bibliography.

**Comment by jsta** [comment]
Looks good to me. Things I noticed:

* Figure 4d looks much improved and closer to the original implementation.

* Several of the figure captions seem to be cross-referencing the wrong figure numbers. In some cases, they reference themselves instead of the preceding figure(s)?

File: code/figure5.jl [hunk 20]
```diff
@@ -0,0 +1,68 @@
+include("seed.jl")
+
+# Simulation with specialist natural enemies with stochasticity on D
+sim1, params1 = simulation(50.0, 25.0, m=0.2, F=4.0, D=0.5, c=1.0, a=0.5, th= 0.0, m=0.5, f=specialist_dyn, D_std=0.5)
```
Comment (ChrisRackauckas) [inline code/figure5.jl]: Some of the function calls are still passing in every parameter, even if it's equal to the default value. This makes it very difficult to see what's actually changed from the baseline value in this line. I am looking for if two parameters were accidentally changed which caused the discrepancy in the figure, but it's hard to tell from this version of the code.

**Comment by ChrisRackauckas** [comment]
A few things:

- Figures 1c, 2b, 3b, 5b, and 6a all have axis limits that obscure the replication a bit. The figure should be fully contained in the plot, and should take up the majority of the plot space. In the first few, seeing the first decimal is crucial to seeing its in the right range.
- I like how Figure 6 (Figure 4 of the paper) is addressed. However, Figure 5  (Figure 3 of the paper)  is not addressed in the discussion section and doesn't look to be in the same range as the paper. There are plenty of excursions to where k>0.5 and the population sizes are about 2x lower than the paper on average. To me, a 50% difference means that figure is still not replicated even accounting for stochasticity (unless the error margins can be show that it's normal to span this range). No evidence was given that this could be accounted for with a different random seed for example. This should be addressed in the discussion. 
- Unnecessary arguments are still passed in some circumstances which obstructs the ability to see which values are changed from the baseline.

**Comment by pdebuyl** [comment]
Hi @jsta @ChrisRackauckas , thank you for the review!

@tpoisot could you address all comments? There is the minor issue of figure numbering but also more specific comments on the stochastic part.

For all, in relation to passing arguments despite them having their default value: we have no "definitive style guide" as part of the [editorial procedure](http://rescience.github.io/edit/) and I believe that is in the interest of the journal, and of the overall effort of replicating scientific findings, to keep some flexibility about coding style.

**Comment by pdebuyl** [comment]
ping @tpoisot

**Comment by tpoisot** [comment]
@DominiqueCaron - can you add discussion of figure 5?

**Comment by pdebuyl** [comment]
Ping @DominiqueCaron ?

**Comment by DominiqueCaron** [comment]
@rougier it is on its way! I already made most of the changes, we just need to update everything accordingly.

**Comment by pdebuyl** [comment]
ping @DominiqueCaron

**Comment by tpoisot** [comment]
@pdebuyl I'm working on the figures, should be done by the end of the day

**Comment by pdebuyl** [comment]
@tpoisot great, thanks :-)

**Comment by pdebuyl** [comment]
ping @tpoisot

**Comment by pdebuyl** [comment]
maybe I should ping on a monday instead of a friday? @tpoisot @DominiqueCaron

**Comment by tpoisot** [comment]
Unless a grant is due on tuesday, yes! It's on my todo-list, rest assured.

**Comment by pdebuyl** [comment]
"happy" grantwriting then, I'll stop pinging for some time.

**Comment by pdebuyl** [comment]
Happy 2019 @tpoisot @DominiqueCaron

A fine year to resubmit :-)

**Comment by rougier** [comment]
Any progress ?

**Comment by tpoisot** [comment]
Still on paternity leave - will work on this when I'm back (by March 10)

**Comment by tpoisot** [comment]
@pdebuyl the PDF is updated following comments by @ChrisRackauckas -- as soon as we are sure that there are no comments left to address, I will clean up the figures. Thanks again for your patience in all that (but life happened in several major ways throughout last year).

**Comment by ChrisRackauckas** [comment]
I took a look at the updated code and it's quite clear. The discussion of the stochastic model differences is a good honest take on what was replicated and where a difference is seen. The only thing I see missing is the axis range change https://github.com/ReScience/ReScience-submission/pull/50#discussion_r199218784 since it's still hard to eyeball how close the replication is, but that's a pretty minor point. Altogether, I think the manuscript is in a good state and fit for publication.

**Comment by pdebuyl** [comment]
@ChrisRackauckas Thanks for the prompt reading.

@tpoisot please update the figures, I'll check the pdf a final time then. You are early w.r.t. your announced date of March 10, no patience was needed :-)

**Comment by tpoisot** [comment]
@pdebuyl Figures updated, and I've built the PDF

**Comment by pdebuyl** [comment]
Hi @tpoisot @DominiqueCaron thanks for the update.

See https://github.com/BIO6032/2018_replication_hassell_1985/pull/30

If you are ok with the changes, I'll follow with the publication. See this throwaway branch https://github.com/pdebuyl/ReScience-submission/tree/caron_2018_edit_with_pdf for a compiled pdf. I'll also have to get rid of the binary pandoc-crossref from the repo...

I could run the code with Julia 1.0.3 (the pdf has these figures) with only a few warnings 

```
This package has been renamed to StatsPlots.
The name StatPlots has been deprecated.

Please run

`]rm StatPlots`
`]add StatsPlots`

Failure to update will mean that you do not receive new developments.

┌ Warning: seriestype density has been moved to StatsPlots.  To use: `Pkg.add("StatsPlots"); using StatsPlots`
└ @ Plots ~/.julia/packages/Plots/UQI78/src/args.jl:1054
```
(last warning repeated 11 more times).

**Comment by pdebuyl** [comment]
Woops, I also forgot to request the DOIs for all references.

**Comment by DominiqueCaron** [comment]
I added most DOIs.
For books, I added the ISBN. For Thompson (1924), I did not find any DOI...

I will let @tpoisot look at it, merge it and update the pdf.

**Comment by tpoisot** [comment]
@pdebuyl all done (I think)

**Comment by pdebuyl** [comment]
Hi @DominiqueCaron @tpoisot thank you, I'll publish this shortly.

**Comment by rougier** [comment]
Gentle reminder

**Comment by pdebuyl** [comment]
Hi @DominiqueCaron @tpoisot I need the ORCID of everyone for the Zenodo archive. I found the one of Timothée Poisot, I would like the three others.

**Comment by pdebuyl** [comment]
I could get by without the ORCID.

This submission has been accepted for publication, and has been published with the DOI [DOI](https://doi.org/10.5281/zenodo.2598793)

**Comment by pdebuyl** [comment]
Hi folks, the paper is online and I am just waiting for the toc of the journal to be updated.

@rougier @khinsen I hereby "Inform the editors-in-chief about the new publication".

**Comment by rougier** [comment]
Thanks. Can you also try to tell the journal editor-in-chief and authors of the original article that it has been replicated ?
